doc/policer_counter/PolicerCounter-HLD.md

```diff
@@ -0,0 +1,331 @@
+15) "SAI_POLICER_STAT_RED_BYTES"
+16) "102400"
+```
+
+Mapping hash table in COUNTERS_DB:
+- "COUNTERS_POLICER_NAME_MAP"
+  - key: POLICER name (e.g: "span_policer")
+  - value: OID of the POLICER counter
+
+```
+; Defines information for COUNTERS_POLICER_NAME_MAP
+    key           = COUNTERS_POLICER_NAME_MAP   ; policer name to its oid
+    ;field        = value
+    <policer_name_to_counter>        = STRING                  ; name of the policer, value is the counter OID
+```
+
+E.g:
+
+```
+127.0.0.1:6379[2]> hgetall COUNTERS_POLICER_NAME_MAP
+ 1) "span_policer"
+ 2) "oid:0x100000000037a"
+```
+### Flows
+
+### POLICER counter registration in FC
+<p align=center>
+<img src="img/policer-flow-add-counter.png" alt="Figure 2. Policer counter registration flow">
+</p>
+
+
+### POLICER counter de-registration in FC
+<p align=center>
+<img src="img/policer-flow-remove-counter.png" alt="Figure 3. Policer counter de-registration flow">
+</p>
+
+### CLI
+
+*policerstat* utility is added to reads counters from the COUNTERS DB using the POLICER table in CONFIG DB and the COUNTERS_POLICER_NAME_MAP. It maps each POLICER to its VID and fetches counter values. If a map entry or VID is missing, it shows N/A, indicating either the POLICER was created without a counter, the policer/map entry isn't created yet, polling is disabled, or syncd hasn't updated the COUNTERS DB.
+
+```
+admin@sonic:~$ policerstat 
+  Name        Green Packets    Green Bytes    Yellow Packets    Yellow Bytes    Red Packets   Red Bytes
+-------      ---------------  -------------  ----------------  --------------- ------------- ------------
+span_policer     600             6144000            300             307200           100           102400
+
+```
+
+Added a new CLI command
+```
+show policer counter <policer_name>
+sonic-clear policer counter <policer_name>
+```
+
+### Configuration and management 
+#### CLI/YANG model Enhancements 
+#### Config DB Enhancements  
+
+Enable POLICER counter polling:
+```
+admin@sonic:~$ counterpoll policer enable
+```
+
+Disable POLICER counter polling (NOTE: POLICER counter objects are still configured in HW):
+```
+admin@sonic:~$ counterpoll policer disable
+```
+
+Set POLICER counter polling interval:
+```
+admin@sonic:~$ counterpoll policer interval [INTERVAL IN MS]
+```
+
+Config DB schema with POLICER key in FLEX COUNTER table:
+
+```json
+{
+    "FLEX_COUNTER_TABLE": {
+        "POLICER": {
+            "FLEX_COUNTER_STATUS": "enable",
+            "POLL_INTERVAL": "10000"
+        }
+    }
+}
+```
+
+Flex Counter YANG model with POLICER group:
+
+```yang
+  container POLICER {
+      /* POLICER_FLEX_COUNTER_GROUP */
+      leaf FLEX_COUNTER_STATUS {
+          type flex_status;
+      }
+      leaf FLEX_COUNTER_DELAY_STATUS {
+          type flex_delay_status;
+      }
+      leaf POLL_INTERVAL {
+          type poll_interval;
+      }
+  }
+```
+		
+### Warmboot and Fastboot Design Impact  
+
+Counter polling is delayed at system startup.

```

Comment:
Can you explain a bit more about how it is delayed at startup? Is this a new thing you are adding? Is it because it uses FC?

Either just for me, or if you think it is worth adding to this section as well.